tests/units/DBmysqlIterator.php

```diff
@@ -717,4 +717,42 @@ function() use($crit) {
          ->isInstanceOf('RuntimeException')
          ->hasMessage('Unknown query operator NOONE');
    }
+
+   public function testUnionQuery() {
+      $union_crit = [
+         ['FROM' => 'table1'],
+         ['FROM' => 'table2']
+      ];
+      $union = new \QueryUnion($union_crit);
+      $raw_query = 'SELECT * FROM (SELECT * FROM `table1` UNION ALL SELECT * FROM `table2`)';
+      $it = $this->it->execute(['FROM' => $union]);
+      $this->string($it->getSql())->isIdenticalTo($raw_query);
+
+      $raw_query = 'SELECT (SELECT * FROM `table1` UNION ALL SELECT * FROM `table2`) FROM `table`';
```

Comment:
It does not seems to be a valid query. If passing `QueryUnion` to `SELECT` is accepted, we should find a real working case.

Reply:
It was a mistake. I've replaced the test with a complex real example of what is expected to work.